### PR TITLE
Add SERIAL_FLOAT_PRECISION option

### PR DESCRIPTION
**Pull request by Fabi0San**
### Description

Introduces DEFAULT_PRINT_FLOAT_PRECISION

### Benefits

Allows increasing the number of fractional digits when printing floats and doubles to serial port.
Effectively increasing the reported precision of probing operations(G30, G38, M48) and configs (M503).

The default setting maintains the current behavior (precision of 2 fractional digits).

I have also removed the calls to FIXFLOAT in G30 as the rounding is already done in string conversion functions.

### CODE REVIEW

**Comment by thinkyhead** [comment]
> I have also removed the calls to FIXFLOAT in G30 as the rounding is already done in string conversion functions.

Float values can sometimes get wack at random. `FIXFLOAT` prevents users seeing "X74.999999" when it should just show "X75" (i.e., as few decimal places as needed) and handles negative values also.

**Comment by Fabi0San** [comment]
> Float values can sometimes get wack at random. `FIXFLOAT` prevents users seeing "X74.999999" when it should just show "X75" (i.e., as few decimal places as needed) and handles negative values also.

I see, can we use a smaller number then?

**Comment by thinkyhead** [comment]
That depends. You only have a certain number of digits in float numbers. While 0.0000005f might be fine when the integer part is small, it will be equivalent to 0.0 once the integer part gains enough digits.

**Comment by Fabi0San** [comment]
I'm reading some commits from 2018 and it seems we removed a lot of FIXFLOAT calls with the assumption that the float to string now does this rounding. I'm checking the implementation now to see if it checks out.

**Comment by Fabi0San** [comment]
MarlinSerial.cpp seems to be handling it, thus making FIXFLOAT unnecessary.

    // Round correctly so that print(1.999, 2) prints as "2.00"
    double rounding = 0.5;
    LOOP_L_N(i, digits) rounding *= 0.1;
    number += rounding;

File: Marlin/src/module/probe.cpp [hunk 1]
```diff
@@ -716,9 +716,9 @@ float Probe::probe_at_point(const float &rx, const float &ry, const ProbePtRaise
       if (stow()) measured_z = NAN;   // Error on stow?
 
     if (verbose_level > 2) {
-      SERIAL_ECHOPAIR_F("Bed X: ", LOGICAL_X_POSITION(rx), 3);
```
Comment (Fabi0San) [inline Marlin/src/module/probe.cpp:719]: This will now default to 2. 
Is that a desired behavior change?

Comment (thinkyhead) [reply Marlin/src/module/probe.cpp:719]: For X position?? Pfft. No one will care, surely.

**Comment by Fabi0San** [comment]
Thank you very much Scott!

**Pull request merged**